src/common.ts

```diff
@@ -100,3 +100,63 @@ export async function validateTxTip(node: GQLNodeInterface, ardriveOracle: ArDri
 
 	return tipPercentage + EPSILON >= 15 && wasValidTipRecipient;
 }
+
+export function daysDiffInEST(prev: Date, curr: Date): number {
+	const prevEstDate = dateToEST(prev);
+	const currEstDate = dateToEST(curr);
+
+	let daysCount = 0;
+	const cursorDate = new Date(prevEstDate.getTime());
+	cursorDate.setHours(0);
+	cursorDate.setMinutes(0);
+	cursorDate.setSeconds(0);
+
+	while (currEstDate.getTime() > cursorDate.getTime()) {
+		const cursorDay = cursorDate.getDate();
+		cursorDate.setDate(cursorDay + 1);
+		if (cursorDate.getTime() < currEstDate.getTime()) {
+			daysCount++;
+		}
+	}
+	return daysCount;
+}
+
+export function weeksDiffInEST(prev: Date, curr: Date): number {
+	const prevEstDate = dateToEST(prev);
+	const currEstDate = dateToEST(curr);
+
+	let weeksCount = 0;
+	const cursorDate = toSunday(prevEstDate);
+
+	while (currEstDate.getTime() > cursorDate.getTime()) {
+		const cursorDay = cursorDate.getDate();
+		const daysPerWeek = 7;
+		cursorDate.setDate(cursorDay + daysPerWeek);
+		if (cursorDate.getTime() < currEstDate.getTime()) {
+			weeksCount++;
+		}
+	}
+	return weeksCount;
+}
+
+function toSunday(date: Date) {
+	const day = date.getDay() || 7;
+	const newDate = new Date(date.getTime());
+	if (day !== 7) {
+		newDate.setHours(-24 * (day - 1));
+	}
+	newDate.setHours(0);
+	newDate.setMinutes(0);
+	newDate.setSeconds(0);
+	return newDate;
+}
```

Comment:
```suggestion
function toSunday(date: Date) {
	const sunday = 0;
	const day = date.getDay();
	const newDate = new Date(date.getTime());

	if (day !== sunday) {
		while (newDate.getDay() > sunday) {
			newDate.setDate(newDate.getDate() - 1);
		}
	}
	newDate.setHours(0);
	newDate.setMinutes(0);
	newDate.setSeconds(0);
	return newDate;
}
```
This change should return the last Sunday :)

Reply:
I simplified that a little bit: 8f3d45f6e969a73718335566c6b4cf9d5a483335
And added tests just to be sure: 53e6f06bb8af00406f4af4979ffd178c528908c6

Reply:
Clever 👍